src/Workspaces/Core/Portable/Indentation/AbstractIndentationService.Indenter.cs

```diff
@@ -171,7 +171,7 @@ public bool TryGetSmartTokenIndentation(out IndentationResult indentationResult)
 
                     var formatter = _service.CreateSmartTokenFormatter(this);
                     var changes = formatter.FormatTokenAsync(Document.Project.Solution.Workspace, token, CancellationToken)
-                                           .WaitAndGetResult(CancellationToken);
+                                           .WaitAndGetResult_CanCallOnBackground(CancellationToken);
```

Comment:
we call this from teh BG.  so this is needed.  i looked into making this all async, but it was a PITA.